packages/plugins/typescript/typescript/src/visitor.ts

```diff
@@ -123,6 +128,39 @@ export class TsVisitor<
     return `Maybe<${super.ListType(node)}>`;
   }
 
+  UnionTypeDefinition(node: UnionTypeDefinitionNode, key: string | number | undefined, parent: any): string {
+    if (this.config.onlyOperationTypes) return '';
+    let withFutureAddedValue: string[] = [];
+    if (this.config.futureProofUnions) {
+      if (this.config.addTypename) {
+        if (this.config.nonOptionalTypename) {
+          withFutureAddedValue = [
+            this.config.immutableTypes ? `{ readonly __typename: "%other" }` : `{ __typename: "%other" }`,
+          ];
+        } else {
+          withFutureAddedValue = [
+            this.config.immutableTypes ? `{ readonly __typename?: "%other" }` : `{ __typename?: "%other" }`,
+          ];
+        }
+      } else {
+        withFutureAddedValue = ['{}'];
```

Comment:
The `__typename` thing isn't really supported on unions, that's why I originally asked about that flag. 
Adding this configuration just because of this new feature might cause confusion for users. Scheme types doesn't have `__typename`, it's something that being resolved only on the operation level.
Maybe this flow should be applied on `typescript-operations` and not here?